File: src/main/java/com/EmosewaPixel/pixellib/resourceAddition/JSONAdder.java
package com.EmosewaPixel.pixellib.resourceAddition;

import com.google.gson.JsonElement;
import net.minecraft.util.ResourceLocation;

import java.util.HashMap;

//This class contains functions that mods can use to add JSONs to the fake data and resource pack
public class JSONAdder {
    protected static HashMap<ResourceLocation, JsonElement> ASSETS = new HashMap<>();
    protected static HashMap<ResourceLocation, JsonElement> DATA = new HashMap<>();

    public static void addAssetsJSON(ResourceLocation location, JsonElement json) {
        ASSETS.put(location, json);
    }

    public static void addDataJSON(ResourceLocation location, JsonElement json) {
        DATA.put(location, json);
    }
}

File: src/main/java/com/emosewapixel/pixellib/resources/JSONAdder.kt
package com.emosewapixel.pixellib.resources

import com.emosewapixel.ktlib.extensions.JsonBuilder
import com.emosewapixel.ktlib.extensions.toResLoc
import com.google.gson.JsonElement
import net.minecraft.util.ResourceLocation

//This class contains functions that mods can use to add JSONs to the fake data and resource pack
object JSONAdder {
    @JvmField
    val ASSETS = hashMapOf<ResourceLocation, JsonElement>()

    @JvmField
    val DATA = hashMapOf<ResourceLocation, JsonElement>()

    @JvmStatic
    fun addAssetsJSON(location: ResourceLocation, json: JsonElement) {
        ASSETS[location] = json
    }

    @JvmStatic
    fun addAssetsJSON(location: ResourceLocation, json: JsonBuilder.() -> Unit) = addDataJSON(location, com.emosewapixel.ktlib.extensions.json(json))

    @JvmStatic
    fun addAssetsJSON(location: String, json: JsonBuilder.() -> Unit) = addAssetsJSON(location.toResLoc(), json)

    @JvmStatic
    fun addDataJSON(location: ResourceLocation, json: JsonElement) {
        DATA[location] = json
    }

    @JvmStatic
    fun addDataJSON(location: ResourceLocation, json: JsonBuilder.() -> Unit) = addDataJSON(location, com.emosewapixel.ktlib.extensions.json(json))

    @JvmStatic
    fun addDataJSON(location: String, json: JsonBuilder.() -> Unit) = addDataJSON(location.toResLoc(), json)
}